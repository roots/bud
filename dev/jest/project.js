/* eslint-disable no-console */
const execa = require('execa')
const {readFile} = require('fs-extra')
const json5 = require('json5')
const {join} = require('path/posix')

/**
 * This class is used to represent an example project being used
 * as the subject of an integration test.
 *
 * @usage
 * ```ts
 *  project = new Project({
 *    name: 'basic',
 *    dir: 'examples/basic',
 *  })

 *  await project.setup()
 *
 *  ...
 *  expect(project.packageJson).toMatchSnapshot()
 *  expect(project.assets['main.js'].length).toBeGreaterThan(10)
 *  ...
 * ```
 *
 * @internal
 */
class Project {
  name

  mode = 'production'

  dist = 'dist'

  storage = '.budfiles'

  assets = {}

  entrypoints = {}

  manifest = {}

  modules = {}

  webpackConfig

  packageJson = {}

  constructor(options) {
    Object.assign(this, {options})

    this.setup = this.setup.bind(this)
    this.projectPath = this.projectPath.bind(this)
    this.publicPath = this.publicPath.bind(this)
    this.readJson = this.readJson.bind(this)
    this.setPackageJson = this.setPackageJson.bind(this)
    this.setManifest = this.setManifest.bind(this)
    this.setAssets = this.setAssets.bind(this)
    this.setEntrypoints = this.setEntrypoints.bind(this)
    this.setWebpackConfig = this.setWebpackConfig.bind(this)
    this.setModules = this.setModules.bind(this)
    this.yarn = this.yarn.bind(this)
  }

  async setup() {
    await this.setPackageJson()
    await this.setManifest()
    await this.setAssets()
    await this.setModules()
    await this.setEntrypoints()
    await this.setWebpackConfig()

    return this
  }

  get dir() {
    return join(`/roots/examples/${this.options.with}`, this.options.name)
  }

  projectPath(file) {
    return `${this.dir}/${file}`
  }

  publicPath(file) {
    return `${this.public}/${file}`
  }

  async readJson(file) {
    const contentString = await readFile(file)
    return json5.parse(contentString)
  }

  async setPackageJson() {
    let packageJson = await this.readJson(this.projectPath('package.json'))

    Object.assign(this, {packageJson})
  }

  async setManifest() {
    let manifest = await this.readJson(
      this.projectPath(`${this.dist}/manifest.json`),
    )

    /**
     * If publicPath is configured, we need to remove from
     * entries or they won't resolve
     */
    manifest = Object.entries(manifest).reduce(
      (a, [k, v]) => ({
        ...a,
        [k]: v.replace(this.public, ''),
      }),
      {},
    )

    Object.assign(this, {manifest})
  }

  async setAssets() {
    const assets = await Object.entries(this.manifest).reduce(
      async (promised, [name, path]) => {
        const assets = await promised
        const buffer = await readFile(
          this.projectPath(`${this.dist}/${path}`),
          'utf8',
        )

        return {
          ...assets,
          [name]: buffer.toString(),
        }
      },
      Promise.resolve(),
    )

    Object.assign(this, {assets})
  }

  async setEntrypoints() {
    try {
      const entrypoints = await this.readJson(
        this.projectPath(`${this.dist}/entrypoints.json`),
      )

      Object.assign(this, {entrypoints})
    } catch (e) {}
  }

  async setWebpackConfig() {
    try {
      const webpackConfig = await readFile(
        this.projectPath(`${this.storage}/bud/webpack.config.js`),
        'utf8',
      )

      this.webpackConfig = webpackConfig
    } catch (e) {}
  }

  async setModules() {
    try {
      const modules = await this.readJson(
        this.projectPath(`${this.storage}/bud/modules.json`),
      )

      Object.assign(this, {modules})
    } catch (e) {}
  }

  async yarn(...opts) {
    const res = execa('yarn', opts, {
      cwd: this.dir,
    })

    await res

    return Promise.resolve()
  }
}

module.exports = Project
