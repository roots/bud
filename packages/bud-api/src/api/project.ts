export const project: Framework.API.Project = function (
  segment,
) {
  return segment
    ? this.disk.get('project')
      ? this.disk.current.get(segment)
      : this.fs.path.join(process.cwd(), segment)
    : this.disk.get('project').base ?? process.cwd()
}
