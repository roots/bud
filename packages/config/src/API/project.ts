export const project: API.Project = function (path) {
  return path
    ? this.fs.path.join(this.store['paths'].get('project'), path)
    : this.store['paths'].project
}
