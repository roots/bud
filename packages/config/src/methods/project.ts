import {Config} from '..'

export const project: Config.Project = function (path) {
  return path
    ? this.fs.path.join(this.store['paths'].get('project'), path)
    : this.store['paths'].get('project')
}
