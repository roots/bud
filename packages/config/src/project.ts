import Bud from '@roots/bud-types'

export const project: Bud.Config.Project = function (path) {
  return path
    ? this.fs.path.join(this.paths.get('project'), path)
    : this.paths.get('project')
}
