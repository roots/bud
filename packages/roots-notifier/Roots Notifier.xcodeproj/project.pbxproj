// !$*UTF8*$!
{
	archiveVersion = 1;
	classes = {
	};
	objectVersion = 46;
	objects = {

/* Begin PBXBuildFile section */
		5199791915B1F92B003AFC57 /* Cocoa.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 5199791815B1F92B003AFC57 /* Cocoa.framework */; };
		5199792315B1F92B003AFC57 /* InfoPlist.strings in Resources */ = {isa = PBXBuildFile; fileRef = 5199792115B1F92B003AFC57 /* InfoPlist.strings */; };
		5199792515B1F92B003AFC57 /* main.m in Sources */ = {isa = PBXBuildFile; fileRef = 5199792415B1F92B003AFC57 /* main.m */; };
		5199792915B1F92B003AFC57 /* Credits.rtf in Resources */ = {isa = PBXBuildFile; fileRef = 5199792715B1F92B003AFC57 /* Credits.rtf */; };
		5199792C15B1F92B003AFC57 /* AppDelegate.m in Sources */ = {isa = PBXBuildFile; fileRef = 5199792B15B1F92B003AFC57 /* AppDelegate.m */; };
		5199792F15B1F92B003AFC57 /* MainMenu.xib in Resources */ = {isa = PBXBuildFile; fileRef = 5199792D15B1F92B003AFC57 /* MainMenu.xib */; };
		5199794215B2F908003AFC57 /* ScriptingBridge.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 5199794115B2F908003AFC57 /* ScriptingBridge.framework */; };
		5199794C15B302F1003AFC57 /* Terminal.icns in Resources */ = {isa = PBXBuildFile; fileRef = 5199794B15B302F1003AFC57 /* Terminal.icns */; };
/* End PBXBuildFile section */

/* Begin PBXFileReference section */
		5199791415B1F92B003AFC57 /* roots-notifier.app */ = {isa = PBXFileReference; explicitFileType = wrapper.application; includeInIndex = 0; path = "roots-notifier.app"; sourceTree = BUILT_PRODUCTS_DIR; };
		5199791815B1F92B003AFC57 /* Cocoa.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Cocoa.framework; path = System/Library/Frameworks/Cocoa.framework; sourceTree = SDKROOT; };
		5199791B15B1F92B003AFC57 /* AppKit.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = AppKit.framework; path = System/Library/Frameworks/AppKit.framework; sourceTree = SDKROOT; };
		5199791C15B1F92B003AFC57 /* CoreData.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = CoreData.framework; path = System/Library/Frameworks/CoreData.framework; sourceTree = SDKROOT; };
		5199791D15B1F92B003AFC57 /* Foundation.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Foundation.framework; path = System/Library/Frameworks/Foundation.framework; sourceTree = SDKROOT; };
		5199792015B1F92B003AFC57 /* Roots Notifier-Info.plist */ = {isa = PBXFileReference; lastKnownFileType = text.plist.xml; path = "Roots Notifier-Info.plist"; sourceTree = "<group>"; };
		5199792215B1F92B003AFC57 /* en */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = en; path = en.lproj/InfoPlist.strings; sourceTree = "<group>"; };
		5199792415B1F92B003AFC57 /* main.m */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.objc; path = main.m; sourceTree = "<group>"; };
		5199792615B1F92B003AFC57 /* Terminal Notifier-Prefix.pch */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = "Terminal Notifier-Prefix.pch"; sourceTree = "<group>"; };
		5199792815B1F92B003AFC57 /* en */ = {isa = PBXFileReference; lastKnownFileType = text.rtf; name = en; path = en.lproj/Credits.rtf; sourceTree = "<group>"; };
		5199792A15B1F92B003AFC57 /* AppDelegate.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = AppDelegate.h; sourceTree = "<group>"; };
		5199792B15B1F92B003AFC57 /* AppDelegate.m */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.objc; path = AppDelegate.m; sourceTree = "<group>"; };
		5199794115B2F908003AFC57 /* ScriptingBridge.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = ScriptingBridge.framework; path = System/Library/Frameworks/ScriptingBridge.framework; sourceTree = SDKROOT; };
		5199794B15B302F1003AFC57 /* Terminal.icns */ = {isa = PBXFileReference; lastKnownFileType = image.icns; path = Terminal.icns; sourceTree = "<group>"; };
		E8DDCF32267EA7B100D52368 /* Base */ = {isa = PBXFileReference; lastKnownFileType = file.xib; name = Base; path = Base.lproj/MainMenu.xib; sourceTree = "<group>"; };
/* End PBXFileReference section */

/* Begin PBXFrameworksBuildPhase section */
		5199791115B1F92B003AFC57 /* Frameworks */ = {
			isa = PBXFrameworksBuildPhase;
			buildActionMask = 2147483647;
			files = (
				5199794215B2F908003AFC57 /* ScriptingBridge.framework in Frameworks */,
				5199791915B1F92B003AFC57 /* Cocoa.framework in Frameworks */,
			);
			runOnlyForDeploymentPostprocessing = 0;
		};
/* End PBXFrameworksBuildPhase section */

/* Begin PBXGroup section */
		5199790915B1F92B003AFC57 = {
			isa = PBXGroup;
			children = (
				5199794B15B302F1003AFC57 /* Terminal.icns */,
				5199794115B2F908003AFC57 /* ScriptingBridge.framework */,
				5199791E15B1F92B003AFC57 /* Terminal Notifier */,
				5199791715B1F92B003AFC57 /* Frameworks */,
				5199791515B1F92B003AFC57 /* Products */,
			);
			sourceTree = "<group>";
		};
		5199791515B1F92B003AFC57 /* Products */ = {
			isa = PBXGroup;
			children = (
				5199791415B1F92B003AFC57 /* roots-notifier.app */,
			);
			name = Products;
			sourceTree = "<group>";
		};
		5199791715B1F92B003AFC57 /* Frameworks */ = {
			isa = PBXGroup;
			children = (
				5199791815B1F92B003AFC57 /* Cocoa.framework */,
				5199791A15B1F92B003AFC57 /* Other Frameworks */,
			);
			name = Frameworks;
			sourceTree = "<group>";
		};
		5199791A15B1F92B003AFC57 /* Other Frameworks */ = {
			isa = PBXGroup;
			children = (
				5199791B15B1F92B003AFC57 /* AppKit.framework */,
				5199791C15B1F92B003AFC57 /* CoreData.framework */,
				5199791D15B1F92B003AFC57 /* Foundation.framework */,
			);
			name = "Other Frameworks";
			sourceTree = "<group>";
		};
		5199791E15B1F92B003AFC57 /* Terminal Notifier */ = {
			isa = PBXGroup;
			children = (
				5199792A15B1F92B003AFC57 /* AppDelegate.h */,
				5199792B15B1F92B003AFC57 /* AppDelegate.m */,
				5199792D15B1F92B003AFC57 /* MainMenu.xib */,
				5199791F15B1F92B003AFC57 /* Supporting Files */,
			);
			path = "Terminal Notifier";
			sourceTree = "<group>";
		};
		5199791F15B1F92B003AFC57 /* Supporting Files */ = {
			isa = PBXGroup;
			children = (
				5199792015B1F92B003AFC57 /* Roots Notifier-Info.plist */,
				5199792115B1F92B003AFC57 /* InfoPlist.strings */,
				5199792415B1F92B003AFC57 /* main.m */,
				5199792615B1F92B003AFC57 /* Terminal Notifier-Prefix.pch */,
				5199792715B1F92B003AFC57 /* Credits.rtf */,
			);
			name = "Supporting Files";
			sourceTree = "<group>";
		};
/* End PBXGroup section */

/* Begin PBXNativeTarget section */
		5199791315B1F92B003AFC57 /* roots-notifier */ = {
			isa = PBXNativeTarget;
			buildConfigurationList = 5199793215B1F92B003AFC57 /* Build configuration list for PBXNativeTarget "roots-notifier" */;
			buildPhases = (
				5199791015B1F92B003AFC57 /* Sources */,
				5199791115B1F92B003AFC57 /* Frameworks */,
				5199791215B1F92B003AFC57 /* Resources */,
			);
			buildRules = (
			);
			dependencies = (
			);
			name = "roots-notifier";
			productName = "Terminal Notifier";
			productReference = 5199791415B1F92B003AFC57 /* roots-notifier.app */;
			productType = "com.apple.product-type.application";
		};
/* End PBXNativeTarget section */

/* Begin PBXProject section */
		5199790B15B1F92B003AFC57 /* Project object */ = {
			isa = PBXProject;
			attributes = {
				LastUpgradeCheck = 1250;
				ORGANIZATIONNAME = "Eloy Dur√°n";
				TargetAttributes = {
					5199791315B1F92B003AFC57 = {
						DevelopmentTeam = FQF57V4Q4F;
						ProvisioningStyle = Automatic;
					};
				};
			};
			buildConfigurationList = 5199790E15B1F92B003AFC57 /* Build configuration list for PBXProject "Roots Notifier" */;
			compatibilityVersion = "Xcode 3.2";
			developmentRegion = en;
			hasScannedForEncodings = 0;
			knownRegions = (
				en,
				Base,
			);
			mainGroup = 5199790915B1F92B003AFC57;
			productRefGroup = 5199791515B1F92B003AFC57 /* Products */;
			projectDirPath = "";
			projectRoot = "";
			targets = (
				5199791315B1F92B003AFC57 /* roots-notifier */,
			);
		};
/* End PBXProject section */

/* Begin PBXResourcesBuildPhase section */
		5199791215B1F92B003AFC57 /* Resources */ = {
			isa = PBXResourcesBuildPhase;
			buildActionMask = 2147483647;
			files = (
				5199792315B1F92B003AFC57 /* InfoPlist.strings in Resources */,
				5199792915B1F92B003AFC57 /* Credits.rtf in Resources */,
				5199792F15B1F92B003AFC57 /* MainMenu.xib in Resources */,
				5199794C15B302F1003AFC57 /* Terminal.icns in Resources */,
			);
			runOnlyForDeploymentPostprocessing = 0;
		};
/* End PBXResourcesBuildPhase section */

/* Begin PBXSourcesBuildPhase section */
		5199791015B1F92B003AFC57 /* Sources */ = {
			isa = PBXSourcesBuildPhase;
			buildActionMask = 2147483647;
			files = (
				5199792515B1F92B003AFC57 /* main.m in Sources */,
				5199792C15B1F92B003AFC57 /* AppDelegate.m in Sources */,
			);
			runOnlyForDeploymentPostprocessing = 0;
		};
/* End PBXSourcesBuildPhase section */

/* Begin PBXVariantGroup section */
		5199792115B1F92B003AFC57 /* InfoPlist.strings */ = {
			isa = PBXVariantGroup;
			children = (
				5199792215B1F92B003AFC57 /* en */,
			);
			name = InfoPlist.strings;
			sourceTree = "<group>";
		};
		5199792715B1F92B003AFC57 /* Credits.rtf */ = {
			isa = PBXVariantGroup;
			children = (
				5199792815B1F92B003AFC57 /* en */,
			);
			name = Credits.rtf;
			sourceTree = "<group>";
		};
		5199792D15B1F92B003AFC57 /* MainMenu.xib */ = {
			isa = PBXVariantGroup;
			children = (
				E8DDCF32267EA7B100D52368 /* Base */,
			);
			name = MainMenu.xib;
			sourceTree = "<group>";
		};
/* End PBXVariantGroup section */

/* Begin XCBuildConfiguration section */
		5199793015B1F92B003AFC57 /* Debug */ = {
			isa = XCBuildConfiguration;
			buildSettings = {
				ALWAYS_SEARCH_USER_PATHS = NO;
				CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED = YES;
				CLANG_CXX_LANGUAGE_STANDARD = "gnu++0x";
				CLANG_ENABLE_OBJC_ARC = YES;
				CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING = YES;
				CLANG_WARN_BOOL_CONVERSION = YES;
				CLANG_WARN_COMMA = YES;
				CLANG_WARN_CONSTANT_CONVERSION = YES;
				CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS = YES;
				CLANG_WARN_EMPTY_BODY = YES;
				CLANG_WARN_ENUM_CONVERSION = YES;
				CLANG_WARN_INFINITE_RECURSION = YES;
				CLANG_WARN_INT_CONVERSION = YES;
				CLANG_WARN_NON_LITERAL_NULL_CONVERSION = YES;
				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = YES;
				CLANG_WARN_RANGE_LOOP_ANALYSIS = YES;
				CLANG_WARN_STRICT_PROTOTYPES = YES;
				CLANG_WARN_SUSPICIOUS_MOVE = YES;
				CLANG_WARN_UNREACHABLE_CODE = YES;
				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
				COPY_PHASE_STRIP = NO;
				ENABLE_STRICT_OBJC_MSGSEND = YES;
				ENABLE_TESTABILITY = YES;
				GCC_C_LANGUAGE_STANDARD = gnu99;
				GCC_DYNAMIC_NO_PIC = NO;
				GCC_ENABLE_OBJC_EXCEPTIONS = YES;
				GCC_NO_COMMON_BLOCKS = YES;
				GCC_OPTIMIZATION_LEVEL = 0;
				GCC_PREPROCESSOR_DEFINITIONS = (
					"DEBUG=1",
					"$(inherited)",
				);
				GCC_SYMBOLS_PRIVATE_EXTERN = NO;
				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
				GCC_WARN_ABOUT_RETURN_TYPE = YES;
				GCC_WARN_UNDECLARED_SELECTOR = YES;
				GCC_WARN_UNINITIALIZED_AUTOS = YES;
				GCC_WARN_UNUSED_FUNCTION = YES;
				GCC_WARN_UNUSED_VARIABLE = YES;
				MACOSX_DEPLOYMENT_TARGET = 10.10;
				ONLY_ACTIVE_ARCH = YES;
				SDKROOT = macosx;
			};
			name = Debug;
		};
		5199793115B1F92B003AFC57 /* Release */ = {
			isa = XCBuildConfiguration;
			buildSettings = {
				ALWAYS_SEARCH_USER_PATHS = NO;
				CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED = YES;
				CLANG_CXX_LANGUAGE_STANDARD = "gnu++0x";
				CLANG_ENABLE_OBJC_ARC = YES;
				CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING = YES;
				CLANG_WARN_BOOL_CONVERSION = YES;
				CLANG_WARN_COMMA = YES;
				CLANG_WARN_CONSTANT_CONVERSION = YES;
				CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS = YES;
				CLANG_WARN_EMPTY_BODY = YES;
				CLANG_WARN_ENUM_CONVERSION = YES;
				CLANG_WARN_INFINITE_RECURSION = YES;
				CLANG_WARN_INT_CONVERSION = YES;
				CLANG_WARN_NON_LITERAL_NULL_CONVERSION = YES;
				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = YES;
				CLANG_WARN_RANGE_LOOP_ANALYSIS = YES;
				CLANG_WARN_STRICT_PROTOTYPES = YES;
				CLANG_WARN_SUSPICIOUS_MOVE = YES;
				CLANG_WARN_UNREACHABLE_CODE = YES;
				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
				COPY_PHASE_STRIP = YES;
				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
				ENABLE_STRICT_OBJC_MSGSEND = YES;
				GCC_C_LANGUAGE_STANDARD = gnu99;
				GCC_ENABLE_OBJC_EXCEPTIONS = YES;
				GCC_NO_COMMON_BLOCKS = YES;
				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
				GCC_WARN_ABOUT_RETURN_TYPE = YES;
				GCC_WARN_UNDECLARED_SELECTOR = YES;
				GCC_WARN_UNINITIALIZED_AUTOS = YES;
				GCC_WARN_UNUSED_FUNCTION = YES;
				GCC_WARN_UNUSED_VARIABLE = YES;
				MACOSX_DEPLOYMENT_TARGET = 10.10;
				SDKROOT = macosx;
			};
			name = Release;
		};
		5199793315B1F92B003AFC57 /* Debug */ = {
			isa = XCBuildConfiguration;
			buildSettings = {
				CODE_SIGN_IDENTITY = "Apple Development";
				CODE_SIGN_STYLE = Automatic;
				COMBINE_HIDPI_IMAGES = YES;
				CURRENT_PROJECT_VERSION = 1;
				DEVELOPMENT_TEAM = FQF57V4Q4F;
				ENABLE_HARDENED_RUNTIME = YES;
				GCC_PRECOMPILE_PREFIX_HEADER = YES;
				GCC_PREFIX_HEADER = "Terminal Notifier/Terminal Notifier-Prefix.pch";
				INFOPLIST_FILE = "Terminal Notifier/Roots Notifier-Info.plist";
				MACOSX_DEPLOYMENT_TARGET = 10.10;
				MARKETING_VERSION = 2.0.1;
				PRODUCT_BUNDLE_IDENTIFIER = "io.roots.roots-notifier";
				PRODUCT_NAME = "roots-notifier";
				PROVISIONING_PROFILE = "";
				PROVISIONING_PROFILE_SPECIFIER = "";
				SDKROOT = macosx;
				WRAPPER_EXTENSION = app;
			};
			name = Debug;
		};
		5199793415B1F92B003AFC57 /* Release */ = {
			isa = XCBuildConfiguration;
			buildSettings = {
				CODE_SIGN_IDENTITY = "Apple Development";
				CODE_SIGN_STYLE = Automatic;
				COMBINE_HIDPI_IMAGES = YES;
				CURRENT_PROJECT_VERSION = 1;
				DEVELOPMENT_TEAM = FQF57V4Q4F;
				ENABLE_HARDENED_RUNTIME = YES;
				GCC_PRECOMPILE_PREFIX_HEADER = YES;
				GCC_PREFIX_HEADER = "Terminal Notifier/Terminal Notifier-Prefix.pch";
				INFOPLIST_FILE = "Terminal Notifier/Roots Notifier-Info.plist";
				MACOSX_DEPLOYMENT_TARGET = 10.10;
				MARKETING_VERSION = 2.0.1;
				PRODUCT_BUNDLE_IDENTIFIER = "io.roots.roots-notifier";
				PRODUCT_NAME = "roots-notifier";
				PROVISIONING_PROFILE = "";
				PROVISIONING_PROFILE_SPECIFIER = "";
				SDKROOT = macosx;
				WRAPPER_EXTENSION = app;
			};
			name = Release;
		};
/* End XCBuildConfiguration section */

/* Begin XCConfigurationList section */
		5199790E15B1F92B003AFC57 /* Build configuration list for PBXProject "Roots Notifier" */ = {
			isa = XCConfigurationList;
			buildConfigurations = (
				5199793015B1F92B003AFC57 /* Debug */,
				5199793115B1F92B003AFC57 /* Release */,
			);
			defaultConfigurationIsVisible = 0;
			defaultConfigurationName = Release;
		};
		5199793215B1F92B003AFC57 /* Build configuration list for PBXNativeTarget "roots-notifier" */ = {
			isa = XCConfigurationList;
			buildConfigurations = (
				5199793315B1F92B003AFC57 /* Debug */,
				5199793415B1F92B003AFC57 /* Release */,
			);
			defaultConfigurationIsVisible = 0;
			defaultConfigurationName = Release;
		};
/* End XCConfigurationList section */
	};
	rootObject = 5199790B15B1F92B003AFC57 /* Project object */;
}
