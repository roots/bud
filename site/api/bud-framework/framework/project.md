---
id: bud-framework.framework.project
title: project property
sidebar_label: project property
hide_title: true
sidebar: "api"
slug: project
---

<!-- Do not edit this file. It is automatically generated by API Documenter. -->

## Framework.project property

Project information and peer dependency management utilities

Signature:

```typescript
project: Project.Interface;
```
