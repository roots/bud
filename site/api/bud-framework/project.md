---
id: bud-framework.project
title: project namespace
sidebar_label: project namespace
hide_title: true
sidebar: "api"
slug: project
---

<!-- Do not edit this file. It is automatically generated by API Documenter. -->

## Project namespace

## Classes

| Class                                           | Description                 |
| ----------------------------------------------- | --------------------------- |
| [Abstract](/api/bud-framework/project/abstract) | Peer service abstract class |

## Interfaces

| Interface                                         | Description            |
| ------------------------------------------------- | ---------------------- |
| [Interface](/api/bud-framework/project/interface) | Peer service interface |
