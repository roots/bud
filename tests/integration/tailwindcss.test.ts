import {beforeAll, describe, it} from '@jest/globals'
import {Project} from '@repo/test-kit/project'

const run = pacman => () => {
  let project: Project

  beforeAll(async () => {
    project = await new Project({
      name: 'tailwindcss',
      dist: 'dist',
      with: pacman,
    }).setup()
  })

  describe('app.css', () => {
    it('is transpiled', () => {
      expect(project.assets['app.css'].includes('@import')).toBeFalsy()
    })

    it('has transpiled @tailwind directives', () => {
      expect(project.assets['app.css'].includes('@apply')).toBe(false)
    })

    it('has whitespace removed', () => {
      expect(project.assets['app.css'].match(/    /)).toBeFalsy()
    })

    it('has breaks removed', () => {
      expect(project.assets['app.css'].match(/\\n/)).toBeFalsy()
    })
  })

  describe('manifest.json', () => {
    it('matches snapshot', () => {
      expect(project.manifest).toMatchSnapshot()
    })
  })
}

describe('tailwindcss', () => {
  describe('npm', run('npm'))
  describe('yarn', run('yarn'))
})
