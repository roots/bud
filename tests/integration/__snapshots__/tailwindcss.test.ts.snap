// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tailwindcss npm manifest.json matches snapshot 1`] = `
Object {
  "app.css": "app.css",
  "app.js": "app.js",
  "index.html": "index.html",
}
`;

exports[`tailwindcss yarn manifest.json matches snapshot 1`] = `
Object {
  "app.css": "app.css",
  "app.js": "app.js",
  "index.html": "index.html",
}
`;
